### PR TITLE
Remove the last mootools dependency from com_finder

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Scope

Use native bootstrap layout instead of funky own style!
Affects Smart Search: Edit Search Filter in admin area
And com_finder advanced in front side
#### Preview

Admin area:
## BEFORE

screen shot 2015-03-05 at 4 32 03
## AFTER

screen shot 2015-03-05 at 4 30 30

Front end:
## BEFORE

screen shot 2015-03-05 at 4 31 47
## AFTER

screen shot 2015-03-05 at 4 59 02
#### Testing:

Admin area:
go to /administrator/index.php?option=com_finder&view=filters
Create a new filter.
play around
Save. Reload etc…

Front end
Create a menu for smart search (component not module)
Enable all the option for advanced search.
Visit that page
Play around with the search!

Report errors or suggestions

### CODE REVIEW

**Comment by losedk** [comment]
Great job! Will do some testing

**Comment by dgt41** [comment]
@chrisdavenport @infograf768 Can I humble ask for some input or some testing here? 
Both of you have a deep understanding how this component work and have a nice testing env, so I will be extremely thankful if you can spare some time for this!

**Comment by Fedik** [comment]
:+1: 

small note:
such big list not very friendly (maybe somehow can make multi columns, or something), and "toggle selection" that near each checkbox very confuse:
screen 2015-03-05 19 11 31 255x878

**Comment by dgt41** [comment]
@Fedik multicolumn or scrollable fixed height will be a better fit here?

**Comment by Fedik** [comment]
@dgt41 not sure what can be a good solution here, but I hate scroll to much :smile:

**Comment by dgt41** [comment]
@Fedik A table style layout is quite hard to be imposed here, as we don’t know the possible width of the every column e.g. a label with some title "something really large to accommodate" 
This means we either hardcode some widths with the possibility that some columns might need some scrolling in x-axis or we hardcore some max-height and then we have some y-axis scrolling.
Or something else? Some front end expert here?

**Comment by dgt41** [comment]
@losedk Can you test this one?

**Comment by losedk** [comment]
@dgt41 Just did some testing:

* The show all button doesn't expand the accordions
* The toggle selection button only selects the first radio button

**Comment by losedk** [comment]
Perhaps the Toggle Selection button above the accordion should be renamed to Select all filters?

**Comment by dgt41** [comment]
@losedk which of the two toggle buttons is not working? (the one above the accordion or the one in each slide?)
Yes Toggle selection should be renamed to something more descriptive.
The show all button seems to toggles the state, thus the behavior is kinda weird. If first slide open this will be closed and all others will become open. I have to find a work around for this...

**Comment by losedk** [comment]
@dgt41 The one above the accordion. 

Just tried again. The show all button does work, but when initially clicking, the first group is collapsed and the others are expanded.

**Comment by dgt41** [comment]
Thank you @losedk for the input! Now all should be fixed! I guess I should have tested this with some real data 😜
screen shot 2015-03-12 at 5 43 41

**Comment by patjun** [comment]
Hi, i tested the patch and it works.

I have 2 suggestions:
- The "Show All" could toogle to "Hide All"
- What Do you think about the number of items behind the Titel of the toogle box. For example "Search by Author (3)"

**Comment by dgt41** [comment]
@patjun Thanks for testing

The button’s function is not toggling all the panels to shown on hidden, is just opening all the panels. The problem is that BS 2.3.2 has same defects here and this is the easiest approach here. But we can make a function to implement this functionality, if people think that this is essential. Bear in mind that as it is right now the code is 1 line, inline in the `onClick` of a button, very simple implementation…

The count is something that can be easily implemented, will push that in a few minutes...

**Comment by designbengel** [comment]
Hello! The "Show all" only opens the first panel. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by dgt41** [comment]
@designbengel Thanks, it should be fine now

**Comment by designbengel** [comment]
Yes, now it´s ok<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by patjun** [comment]
Hi,
i tested the patch again. The numbers in the toggle box are perfect, but i still get problem with the "Show all" Button. 

What i did:
- Open my Filter
- Open Catergory Section -> close Category Section 
- Open Country Section -> close Country Section
- Click "Show all"

Then i get this. All Section are open except the two Categories i opened and closed before. Here is a screenshot.
voila_capture 2015-03-18_06-38-26_nachm

Thanks
Patrick

**Comment by dgt41** [comment]
@patjun Thanks Patrick for testing!
I made some changes to incorporate a bulletproof solution for expanding collapsing the accordion. It looks like this:

On loading the page:
screen shot 2015-03-19 at 5 03 25

On clicking on `Expand all slides` button on the right top corner
screen shot 2015-03-19 at 5 03 30

And then on clicking on `Collapse all slides`
screen shot 2015-03-19 at 5 03 32

This should solve the misbehaving button!

File: administrator/language/en-GB/en-GB.com_finder.ini [hunk 1]
```diff
@@ -105,6 +105,7 @@ COM_FINDER_FILTER_END_DATE_DESCRIPTION="Format YYYY-MM-DD"
 COM_FINDER_FILTER_END_DATE_LABEL="End Date"
 COM_FINDER_FILTER_FIELDSET_DETAILS="Filter Details"
 COM_FINDER_FILTER_FIELDSET_PARAMS="Filter Timeline"
+COM_FINDER_FILTER_HIDE_ALL="Collapse all slides"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_finder.ini]: Just wondering if we really need the word slides. Isn't it enough just to say collape all and expand all

Comment (dgt41) [inline administrator/language/en-GB/en-GB.com_finder.ini]: @brianteeman You are right, it’s pretty obvious what this button will do. Will take them out

**Comment by RickR2H** [comment]
@test played arround using the filters. Patch seems to work.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by wilsonge** [comment]
@test nearly there. Looks good in the frontend. In the backend however if there are no indexed items and you try to create a new search filter (yes I know super unlikely but....) then the buttons still show

image

and actually by clicking on "Expand All" the view starts to fall apart :P 

However other than that one weird test case things look good :) Nice work!!

**Comment by dgt41** [comment]
@test ok! Thanks

**Comment by dgt41** [comment]
I totally missed the front end module, but it should be fine now:
screenshot 2015-06-09 00 53 34


@wilsonge shall I hyde the buttons when items aren't indexed?

**Comment by wilsonge** [comment]
In the backend for the stuff we discussed yes. In the frontend it's really weird. I'd leave it for another PR - because really we probably shouldn't render the module at all or something

**Comment by designbengel** [comment]
A Notice Message would be great for the user - if nothing is indexed -> the search module reports that nothing will be shown then. Same for not enabled plugin...

**Comment by wilsonge** [comment]
It's way outside the scope of this PR tho. Let's get that backend stuff fixed @dgt41 that we talked about at JAB (adding that method into the model) then lets get this tested and merged

**Comment by dgt41** [comment]
@wilsonge done!

**Comment by designbengel** [comment]
#7140 
bildschirmfoto 2015-06-09 um 20 38 15

The filters are overflowing now at the bottom - didn´t found out why yet.

**Comment by dgt41** [comment]
@designbengel I was doing something stupid (inline styling) that’s why you got overflowing modules. Can you retry with the new code?

**Comment by designbengel** [comment]
@dgt41 yes great! now it´s fine - and the other stuff i tested long before successfully :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by roland-d** [comment]
@test All works as described. The module also fits correctly.

@dgt41 The map count in the backend, would be nice if it is updated on-the-fly as items are checked/unchecked. Can you have a look at that?

**Comment by dgt41** [comment]
@roland-d it’s fairly easy, but shall i do it in another PR so we can go ahead and merge this?

**Comment by roland-d** [comment]
@dgt41 Sounds good to me.

**Comment by designbengel** [comment]
@dgt41 No - sorry now i tested the other stuff as well and The "Expand all" Button doesn´t expand all.
When expanding some slides are closed. When open slides manually some remain open, some not. Strange behaviour...
bildschirmfoto 2015-06-09 um 21 59 57
bildschirmfoto 2015-06-09 um 21 59 44

**Comment by designbengel** [comment]
@dgt41 would be great for UX if the checkboxes would check also by clicking on the label. Is that possible?

**Comment by roland-d** [comment]
@dgt41 I can confirm the findings by @designbengel. As for the suggestion for clicking the label, that is a good enhancement.

**Comment by dgt41** [comment]
@designbengel @roland-d what browser?

**Comment by roland-d** [comment]
@dgt41 FF 38.0.5 in my case.

**Comment by dgt41** [comment]
@roland-d can’t replicate it here! can you give me some more info how this is failing?

**Comment by roland-d** [comment]
@dgt41 While I was writing the instructions I can't reproduce it anymore :) Perhaps it was a cache issue. Sorry for the trouble ;)

**Comment by dgt41** [comment]
@roland-d haha spooky code 😜 
I guess if also @designbengel can confirm that this is a cache problem we can move it to rtc

**Comment by designbengel** [comment]
My Browser: Safari 8.0.5 / O SX
cleared the cache -> it worked
Played arround with toggle on/off close/open slides and then the expand all didn´t worked anymore...

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by dgt41** [comment]
@designbengel fixed it

**Comment by designbengel** [comment]
tested successfully :+1: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by zero-24** [comment]
RTC Thanks for testing and coding here :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by roland-d** [comment]
@dgt41 Can you please fix the conflicts so we can go ahead and merge it? Thank you.

**Comment by dgt41** [comment]
@roland-d done!

**Comment by roland-d** [comment]
@dgt41 Not having much luck today ;) Now Travis throws a fit:
```
FILE: ...s/build/joomla/joomla-cms/components/com_finder/helpers/html/filter.php

--------------------------------------------------------------------------------

FOUND 1 ERROR(S) AFFECTING 1 LINE(S)

--------------------------------------------------------------------------------

 372 | ERROR | Please consider a blank line preceding your comment

**Comment by zero-24** [comment]
@roland-d Travis is happy now with the last commit. ;)

**Comment by roland-d** [comment]
@dgt41 Sorry to bother this PR further but I would like to ask some more questions. In your images I see your front-end view gives bootstrapped dropdowns, I don't get them. They are regular dropdowns, is that supposed to be the case?

Second is concerning the backend, when creating a filter, you build the according sliders, inside the slider is the content. The sample data has 24 catgories, there is so much whitespace between each entry that the list gets pretty long. Now imagine on a site with many more categories, that means a scroll craze. Can you have a look at reducing the spacing between each item?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6322">issues.joomla.org/joomla-cms/6322</a>.</sub>

**Comment by dgt41** [comment]
@roland-d front end select tags have no bootstrap touches (the preview is safari browser and should be different in each browser). We could apply chosen there to modify the look and feel but I wouldn’t recommended for the front end. For the input tags inside of the slider a simple css addition should fix the spacing. By the way, which browser you’ve used to test this?

**Comment by roland-d** [comment]
@dgt41 As for front-end, I agree, let's not touch it just was looking to understand the difference between your image and what I saw.

Backend, I am using Firefox on a Mac.

**Comment by dgt41** [comment]
@roland-d It should be ok now

**Comment by roland-d** [comment]
@dgt41 The input height is OK now. Even so, I think we may need to move to chozen at some point, as that is much friendlier with long lists, out of scope for this PR.

**Comment by roland-d** [comment]
This has been merged into 3.5-dev with commit b3ac598 Thanks.

**Pull request closed without merging**